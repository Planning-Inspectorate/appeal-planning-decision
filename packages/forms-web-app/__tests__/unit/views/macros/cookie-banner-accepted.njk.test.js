const nunjucksTestRenderer = require('../nunjucks-render-helper');

describe('views/macros/cookie-banner-accepted', () => {
  it(`should have sensible default attributes`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner-accepted.njk' import cookieBannerAccepted with context -%}
    {{- cookieBannerAccepted() -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting a custom serviceName`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner-accepted.njk' import cookieBannerAccepted with context -%}
    {{- cookieBannerAccepted({ serviceName: "test service name here" }) -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting a custom cookiePagePath`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner-accepted.njk' import cookieBannerAccepted with context -%}
    {{- cookieBannerAccepted({ cookiePagePath: "/some/path/here" }) -}}
    `)
    ).toMatchSnapshot();
  });

  it(`should allow setting all custom attributes`, () => {
    expect(
      nunjucksTestRenderer.renderString(`
    {%- from 'macros/cookie-banner-accepted.njk' import cookieBannerAccepted with context -%}
    {{- cookieBannerAccepted({ cookiePagePath: "/a/b/c", serviceName: "custom service name" }) -}}
    `)
    ).toMatchSnapshot();
  });
});
