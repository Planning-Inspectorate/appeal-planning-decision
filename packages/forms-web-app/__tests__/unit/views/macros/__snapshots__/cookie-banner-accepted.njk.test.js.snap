// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/macros/cookie-banner-accepted should allow setting a custom cookiePagePath 1`] = `
"
  

  

  

  <div class=\\"govuk-cookie-banner govuk-!-display-none\\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on appeal a householder planning decision service\\" data-cy=\\"cookie-banner-accepted\\" id=\\"cookie-banner-accepted\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>You’ve accepted analytics cookies. You can <a class=\\"govuk-link\\" href=\\"/some/path/here\\">change your cookie settings</a> at any time.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"hide\\" type=\\"submit\\" name=\\"cookie_banner_accepted\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accepted-hide-message\\">
              Hide this message
            </button>
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner-accepted should allow setting a custom serviceName 1`] = `
"
  

  

  

  <div class=\\"govuk-cookie-banner govuk-!-display-none\\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on test service name here\\" data-cy=\\"cookie-banner-accepted\\" id=\\"cookie-banner-accepted\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>You’ve accepted analytics cookies. You can <a class=\\"govuk-link\\" href=\\"/cookies\\">change your cookie settings</a> at any time.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"hide\\" type=\\"submit\\" name=\\"cookie_banner_accepted\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accepted-hide-message\\">
              Hide this message
            </button>
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner-accepted should allow setting all custom attributes 1`] = `
"
  

  

  

  <div class=\\"govuk-cookie-banner govuk-!-display-none\\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on custom service name\\" data-cy=\\"cookie-banner-accepted\\" id=\\"cookie-banner-accepted\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>You’ve accepted analytics cookies. You can <a class=\\"govuk-link\\" href=\\"/a/b/c\\">change your cookie settings</a> at any time.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"hide\\" type=\\"submit\\" name=\\"cookie_banner_accepted\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accepted-hide-message\\">
              Hide this message
            </button>
          
        
      </div>
      
    </div>
  
</div>
"
`;

exports[`views/macros/cookie-banner-accepted should have sensible default attributes 1`] = `
"
  

  

  

  <div class=\\"govuk-cookie-banner govuk-!-display-none\\" data-nosnippet role=\\"region\\" aria-label=\\"Cookies on appeal a householder planning decision service\\" data-cy=\\"cookie-banner-accepted\\" id=\\"cookie-banner-accepted\\">
    
    

    <div class=\\"govuk-cookie-banner__message govuk-width-container\\"
    >

    <div class=\\"govuk-grid-row\\">
      <div class=\\"govuk-grid-column-two-thirds\\">
        

        <div class=\\"govuk-cookie-banner__content\\">
    <p>You’ve accepted analytics cookies. You can <a class=\\"govuk-link\\" href=\\"/cookies\\">change your cookie settings</a> at any time.</p>
  </div>
      </div>
    </div>

      
      <div class=\\"govuk-button-group\\">
        
          
            <button value=\\"hide\\" type=\\"submit\\" name=\\"cookie_banner_accepted\\" class=\\"govuk-button\\" data-module=\\"govuk-button\\" data-cy=\\"cookie-banner-accepted-hide-message\\">
              Hide this message
            </button>
          
        
      </div>
      
    </div>
  
</div>
"
`;
